### ISSUE TITLE
pclmulqdq is not detected under -Ctarget-cpu=x86-64-v3 with Miri

### ISSUE THREAD

**Issue by saethlin** [rust-lang/rust#148594]
When I run this code:
```rust
fn main() {
    println!("{}", is_x86_feature_detected!("pclmulqdq"));
    println!("{}", is_x86_feature_detected!("sse4.1"));
}
```
With Miri, enabling target-cpu=x86-64-v3:
```
$ MIRIFLAGS=-Ctarget-cpu=x86-64-v3 cargo +nightly miri run
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.01s
     Running `/home/ben/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/bin/cargo-miri runner target/miri/x86_64-unknown-linux-gnu/debug/scratch`
false
true
```

I'm filing this here because I suspect this is a bug under some `cfg(miri)` code in the standard library. I can't investigate further right now because I'm 3 bugs deep and it's not even lunchtime.
